src/body.rs

```diff
@@ -237,36 +237,42 @@ impl Sender {
                 return Ok(().into());
             }
 
-            try_ready!(tx
-                .as_mut()
-                .expect("tx only taken on error")
-                .poll_ready()
-                .map_err(::error::from));
-
-            if buf.remaining_mut() == 0 {
-                buf.reserve(8192);
-            }
+            if buf.len() == 0 {
```

Comment:
Mind leaving a comment here why we only read more if `len() == 0`? Something explaining that this reads the body before checking `tx.poll_ready`, and if the sender wasn't ready, we don't want to *keep reading more* until it is.

Reply:
Done.